src/components/twilio-voice-basic-call-control/public/twilio-voice-basic-call-control.js

```diff
@@ -22,75 +19,165 @@ class TwilioVoiceBasicCallControl extends HTMLElement {
     });
 
     this.shadowRoot
-      .querySelector('#hold')
-      .addEventListener('click', () => this.#handleHold(true));
-    this.shadowRoot
-      .querySelector('#resume')
-      .addEventListener('click', () => this.#handleHold(false));
-    this.shadowRoot
-      .querySelector('#mute')
-      .addEventListener('click', () => this.#handleMute(true));
-    this.shadowRoot
-      .querySelector('#unmute')
-      .addEventListener('click', () => this.#handleMute(false));
+      .querySelector('#add-conference')
+      .addEventListener('click', () => this.#handleAddConference());
+  }
+
+  async #handleAddConference() {
+    const participant = this.shadowRoot.querySelector('#add-participant').value;
+
+    if (participant === '') {
+      console.log('Please enter a validparticipant');
+    } else {
+      await fetch(
+        `/twilio-voice-basic-call-control/conferences/${
+          this.#conferenceSid
+        }/participants/add/${participant}`,
+        {
+          method: 'POST',
+        }
+      );
+    }
   }
 
   #handleCallMessageReceived(message) {
     const { content, messageType } = message;
     if (messageType === 'user-defined-message') {
-      this.#callSid = content.callSid;
+      const { callSid, conferenceSid, hold, label, muted } = content;
+      this.#participants.set(callSid, { conferenceSid, hold, label, muted });
       this.#conferenceSid = content.conferenceSid;
-      this.#showButtons('#hold-buttons', 'hold');
-      this.#showButtons('#mute-buttons', 'mute');
+      this.#showElement('#add-participant-container', true);
+      this.#renderCallControlButtons();
     }
   }
 
-  async #handleHold(shouldHold) {
-    if (this.#hasParticipantConnected()) {
-      const response = await this.#updateConferenceCall({ hold: shouldHold });
-      if (response.status === 200) {
-        this.#showButtons('#hold-buttons', shouldHold ? 'resume' : 'hold');
-      } else {
-        console.error('Unable to set hold: ', response.error);
-      }
+  async #handleHold(shouldHold, callSid) {
+    const response = await this.#updateConferenceCall(callSid, {
+      hold: shouldHold,
+    });
+    if (response.status === 200) {
```

Comment:
Similar to my above comment, prefer fail-first and return. Less nesting that way.

```
if (response.status !== 200) {
  // handle error case
  return;
}

this.#showElement...
```